### ISSUE TITLE
Setting CoreWebView2EnvironmentOptions.ExclusiveUserDataFolderAccess throws System.NullReferenceException

### ISSUE THREAD

**Issue by mikeduglas** [MicrosoftEdge/WebView2Feedback#2363]
**Description**
Setting CoreWebView2EnvironmentOptions.ExclusiveUserDataFolderAccess throws:

[11360] InitializeAsync() exception System.NullReferenceException: Ññûëêà íà îáúåêò íå óêàçûâàåò íà ýêçåìïëÿð îáúåêòà. 
[11360]    â Microsoft.Web.WebView2.Core.CoreWebView2EnvironmentOptions.set_ExclusiveUserDataFolderAccess(Boolean value) 
[11360]    â claedge.WebView2Ctrl.<InitializeAsync>d__4.MoveNext()

**Version**
SDK: 1.0.1185.39
Runtime: 100.0.1185.39
Framework: WinForms
OS: Win10

**Repro Steps**
```
        async void InitializeAsync(CoreWebView2CreationProperties creationProperties, 
            string additionalBrowserArguments, bool allowSingleSignOnUsingOSPrimaryAccount,
            bool exclusiveUserDataFolderAccess)
        {
            try
            {
                CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions
                {
                    Language = creationProperties?.Language,
                    AdditionalBrowserArguments = additionalBrowserArguments,
                    AllowSingleSignOnUsingOSPrimaryAccount = allowSingleSignOnUsingOSPrimaryAccount,
                    ExclusiveUserDataFolderAccess = exclusiveUserDataFolderAccess    // this throws System.NullReferenceException
                };

                // Create the environment manually
                CoreWebView2Environment env = await CoreWebView2Environment.CreateAsync(
                    creationProperties?.BrowserExecutableFolder,
                    creationProperties?.UserDataFolder,
                    options);

                await webView.EnsureCoreWebView2Async(env);
                OnCoreWebView2InitializationCompleted();
            }
            catch (Exception ex)
            {
                PInvoke.PInvoker.DebugInfo(string.Format("InitializeAsync({0}) exception {1}", additionalBrowserArguments, ex));
            }
        }

```

**Screenshots**

**Additional context**

[AB#39043616](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/39043616)

**Comment by LiangTheDev** [comment]
Could you try to set it after creating the options (options.ExclusiveUserDataFolderAccess = exclusiveUserDataFolderAccess) and see whether that works?

**Comment by LiangTheDev** [comment]
Actually, no need to try it out, it is a bug in WebView2 SDK.

**Comment by mikeduglas** [comment]
Yup, same exception.

--
Mike


чт, 14 апр. 2022 г. в 00:54, Liang Zhao - MSFT ***@***.***>:

**Comment by LiangTheDev** [comment]
The bug is now fixed internally. We would have to wait for next SDK release to get the fix though.

**Comment by champnic** [comment]
This is fixed in 1.0.1210.30 SDK and 1.0.1243-prerelease SDK. Thanks!
(Note: These SDKs has been deprecated due to compat issues with the StatusBarTextChanged API, but should be safe to use if you don't use that API. See here for more info: https://github.com/MicrosoftEdge/WebView2Feedback/issues/2414#issuecomment-1116335991)
